### PR TITLE
FINERACT-1724-Snooze-fee-due-date-payment-due-new-fee-failure

### PR DESCRIPTION
**Pull request by ruchiD**
## Description
Fix 
Integration test

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanAccrualWritePlatformServiceImpl.java [hunk 1]
```diff
@@ -385,10 +385,9 @@ private void updateChargeForSubmittedOnDate(Collection<LoanChargeData> chargesDa
         LocalDate scheduleEndDate = accrualData.getDueDateAsLocaldate();
         for (LoanChargeData loanCharge : chargesData) {
             BigDecimal chargeAmount = BigDecimal.ZERO;
-            if (((accrualData.getInstallmentNumber() == 1 && loanCharge.getSubmittedOnDate().isEqual(startDate))
-                    || loanCharge.getSubmittedOnDate().isBefore(startDate) || loanCharge.getSubmittedOnDate().isEqual(startDate)
-                    || loanCharge.getSubmittedOnDate().isAfter(startDate)) && !loanCharge.getSubmittedOnDate().isAfter(endDate)
-                    && !loanCharge.getDueDate().isBefore(startDate) && !loanCharge.getDueDate().isAfter(scheduleEndDate)) {
+            if (((accrualData.getInstallmentNumber() == 1 && loanCharge.getSubmittedOnDate().isEqual(startDate)
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanAccrualWritePlatformServiceImpl.java:388]: can you explain this conditions to me? i am not sure i understand fully the logic behind it... :/

Comment (ruchiD) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanAccrualWritePlatformServiceImpl.java:388]: This condition was to check if charge is submitted and is due on disbursement date.

Comment (adamsaghy) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanAccrualWritePlatformServiceImpl.java:388]: I am still a little confused.
**Based on the condition here**:
if it is the first installment AND the submitted on date is equals with installment start date AND the due date of the charge is equals with the start date 
OR it is after the start date of installment period 
AND the submitted on date is before or equals with end date of installment AND due date is before or equals with **schedule end date** (not with installment end date)

**If the accrual calculation is based on due date the condition is the following:**
if it is the first installment AND due date is equals with start date
OR
due date is after the start date AND before or equals with installment end date 

Can you help me to understand the why does it need to be different and if the above conditions are right, then please help me to get understand it further?

**Review by adamsaghy — APPROVED** [review]
LGTM

**Pull request merged**